tests/e2e/transaction/transaction_submission.test.ts

```diff
@@ -3,100 +3,362 @@
 
 import { Account, AptosConfig, Network, Aptos, U64, SigningScheme } from "../../../src";
 import { waitForTransaction } from "../../../src/internal/transaction";
-import { FUND_AMOUNT, longTestTimeout } from "../../unit/helper";
+import { longTestTimeout } from "../../unit/helper";
+import { fundAccounts, publishTransferPackage, singleSignerScriptBytecode, multiSignerScriptBytecode } from "./helper";
 
 describe("transaction submission", () => {
-  describe("submitTransaction", () => {
-    test(
-      "it submits a script transaction",
-      async () => {
-        const config = new AptosConfig({ network: Network.LOCAL });
-        const aptos = new Aptos(config);
-        const alice = Account.generate();
-        await aptos.fundAccount({ accountAddress: alice.accountAddress.toString(), amount: FUND_AMOUNT });
-        const bob = Account.generate();
-        await aptos.fundAccount({ accountAddress: bob.accountAddress.toString(), amount: FUND_AMOUNT });
-        const rawTxn = await aptos.generateTransaction({
-          sender: alice.accountAddress.toString(),
-          secondarySignerAddresses: [bob.accountAddress.toString()],
+  const config = new AptosConfig({ network: Network.LOCAL });
+  const aptos = new Aptos(config);
+  const senderAccount = Account.generate();
+  const recieverAccounts = [Account.generate(), Account.generate()];
+  const senderSecp256k1Account = Account.generate(SigningScheme.Secp256k1Ecdsa);
+  const secondarySignerAccount = Account.generate();
+  const feePayerAccount = Account.generate();
+  beforeAll(async () => {
+    await fundAccounts(aptos, [
+      senderAccount,
+      senderSecp256k1Account,
+      ...recieverAccounts,
```

Comment:
```suggestion
      ...receiverAccounts,
```

ctrl+f `reciever` nit :D